### ISSUE TITLE
Print important data on top when running

### ISSUE THREAD

**Issue by AbdealiJK** [coala-analyzer/coala#2044]
As we have so many problems with coala, coala-bears and stuff, we should print:
 - python version
 - pip version
 - coala version
 - Version of every module which has the `coalabears` entry point

We could also print (but not important for this issue. Make a new issue for these):
 - Number of files we will be parsing
 - Time taken to run (in the footer) - like pytest

This is similar to how pytest prints a header. It'd be very useful for debugging !

**Comment by SanketDG** [comment]
So print all of this out in `[DEBUG]` mode?

**Comment by AbdealiJK** [comment]
I'd say print it for everything (maybe `[INFO]` ?)
Or just a `ConsolePrinter` (without a log_printer) as this isn't really a Log

Its just good for debugging and doesn't interfere much (loads of softwares have headers, python shell, ipython, R, julia, pytest, etc. for example)

**Comment by sils1297** [comment]
I'm for using the debug channel, it's debug info after all

**Comment by Makman2** [comment]
definitely debug^^

**Comment by AbdealiJK** [comment]
Adding versions of python, etc was easy. But adding for the detected bears using entrypoints is not as easy.

Because that happens in `Section.bear_dirs()` which does not have a log_printer.
Also, Sectionbear_dirs finds the entry points for every section - which is not required. We should do this only once.

**Comment by Makman2** [comment]
we damn need the built-in logging facilities from python...

**Comment by ujas800** [comment]
do u want like this.....?



(venv) shri@Ujas"___":~/coala-tutorial$ coala --files=1.py --bears=SpaceConsistencyBear --save
**********************************************
python version :: 3.4.3
pip version :: 8.1.2
coala version :: 0.3.0.dev0
**********************************************
Executing section Default...

1.py
|   2| ••••••••--->print("hello")
|    | [NORMAL] SpaceConsistencyBear:
|    | Line contains following spacing inconsistencies:
|    | - Tabs used instead of spaces.
|----|    | /home/shri/coala-tutorial/1.py
|    |++++| /home/shri/coala-tutorial/1.py
|   1|   1| def main():
|   2|    |-        	print("hello")
|    |   2|+            print("hello")
|   3|   3|             print("bye")
|   4|   4| 
|   5|   5| if __name__=='__main__':main()
|    | The following actions are applicable to this result:
|    |  0: Apply no further actions.
|    |  1: Open the affected file(s) in an editor.
|    |  2: Apply the patch automatically.
|    | Please enter the number of the action you want to execute (Ctrl-D to exit).

**Comment by ujas800** [comment]
do u want like this...?

http://pastebin.com/yMdae6rE

**Comment by sils1297** [comment]
Did somebody delete comments here?

**Comment by Makman2** [comment]
Don't think so^^

**Comment by sils1297** [comment]
I did get emails for commments that aren't there anymore

**Comment by cylussec** [comment]
I would be willing to look at this. I noticed in a cursory look that 'console_printer' is used. Are you suggesting to move to using the python built in logging and have the different logging levels instead?

**Comment by sils** [comment]
Please use builtin logging and not LogPrinters, we're phasing those out. Should I assign you?

**Comment by cylussec** [comment]
Sure!

**Comment by cylussec** [comment]
Committed some work at https://github.com/cylussec/coala/commit/3fa3f9b6bbe3c0a2086fdc4bf5e470b5f55696fb so people can make sure I am going the right direction. Its not ready yet... it doesnt handle colors (should be done in the format function) and there are some duplicate log messages that I need to figure out. But the major work has been changed.

**Comment by sils** [comment]
Hey @cylussec I'm not fully sure about this; you're switching some prints (normal output) to log output - is that really a wise idea?

I feel like you have taken on a much larger task than this, maybe this was a misunderstanding? *This issue* is purely about https://github.com/coala/coala/issues/2044#issue-150646267 - and to do that I think you can simply

```python
import logging
...
logging.debug(...)
```

with no further ties attached.

As you probably know we're working on getting rid of the log_printer objects everywhere, but this doesn't need to be done here (otherwise this wouldn't be difficulty/low). Ideally instead of passing around the log_printer objects the functions would just simply use logging.

I hope that helps?

**Comment by cylussec** [comment]
So I might have been taking your last comment about using built in logging and ran with it. With python's logging, we are able to write to stdout and log files at the same time, which is why I converted them all. Is that not what we wanted?

And I probably took on more than this issue was. I kind of got going on it, and just kept running with it. 

Although, since I have already done a lot of work for this, should we just keep pushing foward on this?

**Comment by sils** [comment]
I think it's important to keep the issues separated. This one is about printing more debug information at the beginning, that should be pretty easy and not involve any core changes like converting any printing functions.

The second issue is that we use LogPrinter objects all over the place which we deprecated in favour of builtin logging. That does *not* affect the ConsolePrinter since that one's used to print to the console with colors and stuff - but the LogPrinters can easily be removed and instead we can just import logging and do logging.whatever. This is not this issue.

**Comment by cylussec** [comment]
I am having trouble figuring out how to get the version of every module which has the coalabears entry point. Where is that? I have the rest of this one done.

**Comment by sils** [comment]
I think getting the version of every entry point might be a huge performance hit, not sure if we really want that unless absolutely needed. There's a function get_registered_bear_dirs I believe, just grep fit entrypoint probably gets you to wherever we resolve then

**Comment by AbdealiJK** [comment]
Were already doing bear.__file__ so we can just add bear.__version__ to it as well. Its not a major performance hit.
Entry point finding was in collections.py iirc

**Comment by cylussec** [comment]
I am not seeing a collections.py file, or any other way to get that info. Am I missing it?

**Comment by sils** [comment]
he probably means `coalib/collecting/Collectors.py`?

**Comment by AbdealiJK** [comment]
Yeah, sorry was just giving a high level picture.

**Comment by jayvdb** [comment]
> Check version conflicts using https://pypi.python.org/pypi/pip-conflict-checker

This is an important type of feature, and IMO is big enough that it should be a separate task.

**Comment by KVGarg** [comment]
I would like to take up this issue. Can you help me in clearing some of my doubts?
1. Does this mean that we have to print versions of each bear?
    > Version of every module which has the coalabears entry point
2. Do all this infomation be need to be printed in DEBUG mode?
3. And, if we need to print version of each bear (from point 1), then how can we get it? because AFAIK we don't add/set attribute for a coala-bear version!

**Comment by KVGarg** [comment]
@li-boxuan @Makman2 Does this issue still needs some work? because currently `python, coala and system` information is being logged when running in DEBUG mode
